src/instructlab/sdg/pipeline.py

```diff
@@ -67,17 +67,33 @@ def generate(self, dataset) -> Dataset:
         dataset: the input dataset
         """
         for block_prop in self.chained_blocks:
+            # Parse and instantiate the block
             block_name = block_prop["name"]
             block_type = _lookup_block_type(block_prop["type"])
             block_config = block_prop["config"]
             drop_columns = block_prop.get("drop_columns", [])
             drop_duplicates_cols = block_prop.get("drop_duplicates", False)
             block = block_type(self.ctx, self, block_name, **block_config)
-
             logger.info("Running block: %s", block_name)
             logger.info(dataset)
 
-            dataset = block.generate(dataset)
+            # Execute the block and wrap errors with the block name/type
+            try:
+                dataset = block.generate(dataset)
+
+            except Exception as err:
+                block_exc_err = (
```

Comment:
What would be so wrong with simply doing

```
raise BlockGenerateError(f"Error occurred in block.generate() '{block.name}': {str(err)}") from err
```

Reply:
Good question. I was trying to be minimally invasive in case there was type-specific error handling somewhere above here in the CLI, but I think that's probably just being overly cautious. I'll switch to a well-defined error like this.